python/ray/data/tests/test_dataset_pipeline.py

```diff
@@ -814,8 +814,12 @@ def consume(pipe, owned_by_consumer):
     ray.get([consume.remote(splits[0], True), consume.remote(splits[1], True)])
 
 
-# Run at end of file to avoid segfault https://github.com/ray-project/ray/issues/31145
 def test_incremental_take(shutdown_only):
+    # TODO(https://github.com/ray-project/ray/issues/31145): re-enable
+    # after the segfault bug is fixed.
+    if DatasetContext.get_current().new_execution_backend:
+        return
```

Comment:
Move to a separate file instead of disabling.

Reply:
That doesn't work either for this test either, bazel still failed it.

Reply:
I think if you move this into a separate file, you can also remove the ray.shutdown() code below. That should avoid restarting Ray entirely.

Reply:
It failed (`bazel test python/ray/data:tests/test_pipeline_incremental_take` locally) still, but the CI test (which uses bazel) passed, didn't find out why.

Reply:
Yes, you need to remove the ray.shutdown/init on line 823/824.

Reply:
Maybe there was another Ray running locally or something.